presto-main/src/main/java/com/facebook/presto/testing/MaterializedResult.java

```diff
@@ -419,6 +419,9 @@ public static MaterializedResult materializeSourceDataStream(ConnectorSession se
             if (outputPage == null) {
                 break;
             }
+            if (outputPage.getPositionCount() == 0) {
+                continue;
```

Comment:
should this be also break and combine with above if

Reply:
@bhhari `null` page means there is no more data; empty page means that all the data in a batch was filtered out, but there still might be more batches.